text/profiles/0237-profiles-data-model.md

```diff
@@ -0,0 +1,1015 @@
+TODO: add a field that indicates the format of the original payload?
+
+##### Field `stacktraces`
+
+A lookup table of Stacktraces. Other messages refer to Stacktraces in this table by index.
+
+##### Field `mappings`
+
+A lookup table of Mappings. Other messages refer to Mappings in this table by index.
+
+##### Field `locations`
+
+A lookup table of Locations. Other messages refer to Locations in this table by index.
+
+##### Field `functions`
+
+A lookup table of Functions. Other messages refer to Functions in this table by index.
+
+##### Field `links`
+
+A lookup table of Links to trace spans associated with this profile. Other messages refer to Links in this table by index. The first message must be an empty Link — this represents a null Link.
+
+##### Field `attribute_sets`
+
+A lookup table of AttributeSets. Other messages refer to AttributeSets in this table by index. The first message must be an empty AttributeSet — this represents a null AttributeSet.
+
+##### Field `string_table`
+
+A lookup table of strings. Other messages refer to strings in this table by index.
+The 0-th element must be an empty string ("").
+
+##### Field `profile_types`
+
+List of profile types included in this profile. The first item in the list is considered to be the "default" profile type. Example profile types are allocated objects or allocated bytes.
+</details>
+
+#### Message `ProfileType`
+
+Represents a single profile type. It implicitly creates a connection between Stacktraces, Links, AttributeSets, values and timestamps. The connection is based on the order of the elements in the corresponding lists. This implicit connection creates an ephemeral structure called Sample. The length of reference lists must be the same. It is acceptable however for timestamps, links and attribute set lists to be empty. It is not acceptable for stacktrace or values lists to be empty.
+
+<details>
+<summary>Field Descriptions</summary>
+
+##### Field `aggregation_temporality`
+
+aggregation_temporality describes if the aggregator reports delta changes
+since last report time, or cumulative changes since a fixed start time.
+
+##### Field `sample_rate`
+
+Profiler sample rate in Hz. This parameter indicates the frequency at which samples are collected, specifically for CPU profiles. Common values are 99 or 100. [Optional].
+
+##### Field `type_index`
+
+Index into the string table for the type of the sample. Example values are "cpu", "alloc_objects", "alloc_bytes", "block_contentions". Full list is defined in [semantic-conventions](https://github.com/open-telemetry/semantic-conventions)
+
+##### Field `unit_index`
+
+Index into the string table for the unit of the sample. Example values are "ms", "ns", "samples", "bytes". Full list is defined in [semantic-conventions](https://github.com/open-telemetry/semantic-conventions)
```

Comment:
Are we using [UCUM](https://ucum.org/ucum) for these units like we do for metrics?